File: Laba2/Laba2/ITransport.cs
﻿using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba2
{
    public interface ITransport
    {
        int bodyHeight { get; }

        Point Center { get; }

        Bitmap Pict { get; }

        void move(Graphics g);

        void draw(Graphics g);

        void setPosition(int x, int y);

        void loadPassenger(int count);

        int getPassenger();

        void setMainColor(Color color);

        void setDopColor(Color dopColor);

        string getInfo();
    }
}

File: Laba2/src/main/ITransport.java
package main;

import java.awt.*;
import java.awt.image.BufferedImage;

public interface ITransport {

    void move(Graphics g);

    void draw(Graphics g);

    void setPosition(int x, int y);

    void loadPassenger(int count);

    int getPassenger();
    
    int getBodyHeight();
    
    BufferedImage Pict();
    
    Point Center();


}
